v3/services/zip_file_generator.rb

```diff
@@ -0,0 +1,50 @@
+require 'zip'
+
+class ZipFileGenerator
```

Comment:
What do you think of the name `Archive` or `ZipArchive`? We don't need the `Generator` suffix.